includes/class-wc-payments-utils.php

```diff
@@ -1084,7 +1084,7 @@ public static function is_store_api_request(): bool {
 			$rest_route = sanitize_text_field( $_REQUEST['rest_route'] ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash,WordPress.Security.NonceVerification
 		} else {
 			$url_parts    = wp_parse_url( esc_url_raw( $_SERVER['REQUEST_URI'] ?? '' ) ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash
-			$request_path = rtrim( $url_parts['path'], '/' );
+			$request_path = $url_parts ? rtrim( $url_parts['path'], '/' ) : '';
```

Comment:
An empty `$request_path` leads to `$rest_route` being set to an empty string too, which fails the pattern matching in the foreach block below leading to the method returning false as expected.

I have added a unit test just in case.